src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -6083,7 +6141,7 @@ private static bool DoDnsCallback(IAsyncResult result, MultipleAddressConnectAsy
             return DoMultipleAddressConnectCallback(PostOneBeginConnect(context), context);
         }
 
-        private class MultipleAddressConnectAsyncResult : ContextAwareResult
+        private sealed class MultipleAddressConnectAsyncResult : ContextAwareResult
```

Comment:
> What is the reason to seal the class?

It's a private type that AFAICT is not an extension point.

> During work on NetworkInformation I noticed the same changes that had to be reverted.

What caused the reversion? Was there code outside of NetworkInformation that was inheriting from the type?

> https://msdn.microsoft.com/en-us/library/ms229023(v=vs.110).aspx
X DO NOT seal classes without having a good reason to do so.

I believe that this guidance is out of date. @stephentoub ?